rules/Php73/Rector/FuncCall/JsonThrowOnErrorRector.php

```diff
@@ -49,24 +52,48 @@ public function getRuleDefinition(): RuleDefinition
      */
     public function getNodeTypes(): array
     {
-        return [FuncCall::class];
+        return [StmtsAwareInterface::class];
     }
 
     /**
-     * @param FuncCall $node
+     * @param StmtsAwareInterface $node
      */
     public function refactor(Node $node): ?Node
     {
-        if ($this->shouldSkip($node)) {
+        // if found, skip it :)
+        $hasJsonErrorFuncCall = (bool) $this->betterNodeFinder->findFirst(
+            $node,
+            fn (Node $node): bool => $this->isNames($node, ['json_last_error', 'json_last_error_msg'])
+        );
+
+        if ($hasJsonErrorFuncCall) {
             return null;
         }
 
-        if ($this->isName($node, 'json_encode')) {
-            return $this->processJsonEncode($node);
-        }
+        $this->hasChanged = false;
```

Comment:
Rector rule is shared service, so `$this->hasChanged` should not be used, as it marking all next skipped this rule as changed, local `$hasChanged` should be used instead .

Reply:
This should be ok, as `hasChanged` is local private property only in current Rector rule.

Reply:
on same Rector re-called, unfortunatelly, `$this->hasChanged` still alive on next same rector rule call

Reply:
Not sure I see it. Could you add a failing test fixture?

Reply:
this is happen when:

- same rector called 
- jump to different file

Reply:
that's need e2e test as the change is about showing:

```
Applied rules:
 * JsonThrowOnErrorRector
```

Reply:
Oh, I see, `$this->hasChanged = false` always init inside `refactor()`, that's why it is working as expected.

The not working part is when we define `private bool $hasChanged = false` without re-init in `refactor()` method.

so, this change is fine then ;)

Reply:
Good to double-check though :), thank you :+1:

Reply:
I created PR https://github.com/rectorphp/rector-src/pull/4454 to apply init `$this->hasChanged = false;` on `refactor()` method.